### PR TITLE
contractsdk/cpp: add builtin hash function

### PR DESCRIPTION
**Pull request by icexin**
Add builtinResolver to add xvm builtin functions.

### CODE REVIEW

File: core/contractsdk/cpp/xchain/builtins.cc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "xchain/builtins.h"
+#include <string>
+
+extern "C" void xvm_hash(const char* name, const char* inputptr, int inputlen,
+                         char* outputptr, int outputlen, int hex_encode);
+
+namespace xchain {
+std::string sha256(const std::string& input) {
```
Comment (yucaowang) [inline core/contractsdk/cpp/xchain/builtins.cc]: hash/crypto functions could be aggregated in a sub namespace?

File: core/contractsdk/cpp/xchain/builtins.cc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    char out[64];
+    xvm_hash("sha256", &input[0], input.size(), out, sizeof(out), 1);
+    return std::string(out, sizeof(out));
+}
+
+std::string sha256raw(const std::string& input) {
```
Comment (yucaowang) [inline core/contractsdk/cpp/xchain/builtins.cc]: Another implementation way could be providing a bytes to hex string utility, and keep hash return bytes.

File: core/contractsdk/cpp/xchain/builtins.cc (hunk 1 above)
Comment (icexin) [reply core/contractsdk/cpp/xchain/builtins.cc]: Makes sense, has been added.

File: core/contractsdk/cpp/xchain/builtins.cc (hunk 2 above)
Comment (icexin) [reply core/contractsdk/cpp/xchain/builtins.cc]: hex_encode and hex_decode have been added

**Pull request merged**